pages/gallery/_.vue

```diff
@@ -54,21 +65,35 @@
           return {
             description: response.items[0].fields.description,
             images: response.items[0].fields.hasPart,
-            title: response.items[0].fields.name
+            title: response.items[0].fields.name,
+            canonicalURL: null
           };
         })
         .catch((e) => {
           error({ statusCode: 500, message: e.toString() });
         });
     },
+    computed: {
+      shareMediaUrl() {
+        if (this.images.length <= 0) return null;
+        if (!this.images[0].fields.thumbnailUrl) return null;
+        return this.images[0].fields.thumbnailUrl;
+      }
+    },
+    mounted() {
+      this.canonicalURL = window.location.href.split(/\?|#/)[0];
+    },
     head() {
       return {
         title: this.title,
         meta: [
           { hid: 'title', name: 'title', content: this.title },
           { hid: 'description', name: 'description', content: this.description },
           { hid: 'og:title', property: 'og:title', content: this.title },
-          { hid: 'og:description', property: 'og:description', content: this.description }
+          { hid: 'og:description', property: 'og:description', content: this.description },
```

Comment:
If there is no description, the description is "undefined". Let's leave it blank instead.